services/management/ia/templates_service.go

```diff
@@ -17,15 +17,134 @@
 package ia
 
 import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"path/filepath"
+
+	saas "github.com/percona-platform/saas/pkg/alert"
 	iav1beta1 "github.com/percona/pmm/api/managementpb/ia"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	shippedRuleTemplatePath     = "/tmp/ia1/*.yml"
+	userDefinedRuleTemplatePath = "/tmp/ia2/*.yml"
 )
 
+// TemplatesService is responsible for interactions with IA rule templates.
 type TemplatesService struct {
-	iav1beta1.UnimplementedTemplatesServer // TODO remove
+	l                           *logrus.Entry
+	shippedRuleTemplatePath     string
+	userDefinedRuleTemplatePath string
+	rules                       map[string]saas.Rule
```

Comment:
do we expect `collectRuleTemplates` to be called concurrently? should I add a lock here?